### PR TITLE
Update the anim state graph to support the new data schema

### PR DESCRIPTION
**Pull request by ellt92**
I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/components/anim/state-graph.js [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
-            });
-        } else {
-            this._layers = data.layers;
-        }
-        this._parameters = {};
-        for (var paramId in data.parameters) {
-            var param = data.parameters[paramId];
-            this._parameters[param.name] = { type: param.type, value: param.value };
+    var i;
+    if (!Array.isArray(data.layers)) {
```
Comment (slimbuck) [inline src/framework/components/anim/state-graph.js:5]: Why this test here? Is data.layers sometimes an array? Shouldn't it be testing whether data.layers is an object instead (as the code below assumes)?

File: src/framework/components/anim/state-graph.js [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
-                transitions: data.transitions
-            });
-        } else {
-            this._layers = data.layers;
-        }
-        this._parameters = {};
-        for (var paramId in data.parameters) {
-            var param = data.parameters[paramId];
-            this._parameters[param.name] = { type: param.type, value: param.value };
+    var i;
```
Comment (slimbuck) [inline src/framework/components/anim/state-graph.js:4]: Might be good to document what data is in this case?

File: src/framework/components/anim/state-graph.js (hunk 1 above)
Comment (ellt92) [reply src/framework/components/anim/state-graph.js:5]: Yeah, I missed out the assignment of data.layers in the case it's an array. I've added this in now.

File: src/framework/components/anim/state-graph.js (hunk 2 above)
Comment (ellt92) [reply src/framework/components/anim/state-graph.js:4]: 👍

**Pull request merged**